viewer/packages/tools/src/Timeline/TimelineTool.test.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+import { Cognite3DModel } from '@reveal/core';
+import { CogniteClient } from '@cognite/sdk';
+import { CadMaterialManager, CadNode } from '@reveal/rendering';
+import { CadModelMetadata } from '@reveal/cad-parsers';
+import { createCadModelMetadata, generateSectorTree } from '../../../../test-utilities';
+import { NodesLocalClient } from '@reveal/nodes-api';
+
+import { IndexSet } from '../../../../packages/utilities';
+import { TreeIndexNodeCollection } from '../../../../packages/cad-styling';
+import nock from 'nock';
+import { TimelineTool } from './TimelineTool';
+import TWEEN from '@tweenjs/tween.js';
+
+describe('TimelineTool', () => {
```

Comment:
Test is getting decent, but ask yourself the following for each of the tests:
1. Does the test one or several conditions? If several, consider if it would make sense to split into several tests.
2. Can more test setup be moved to `beforeEach`? E.g. can the TimelineTool-initialization and keyframe generation be moved here? Maybe it's possible to create a utility function for creating the tool?
3. Is the test name descriptive? I consider a good test name to 1. describe relevant conditions (how stuff has been initialised), 2. clearly state what functionality you test (e.g. `play()`) and 3. expected condition (e.g. 'loops through all keyframes in order).
4. If helper-classes should have their own tests - e.g. it might make sense to test Keyframe separately.

Note that this isn't necessarily a list of things I think should be changed, but a list I think you should consider.